File: 12_linked_list_sum-list/sumList.js
class Node {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}
// My solution
const sumList = (head) => {
  let current_node = head
  let sum = 0
  while(current_node){
    sum += current_node.val
    current_node = current_node.next
  }
  return sum
};


// Recursive Solution

const sumList = (head) => {
  if (head === null) return 0;
  return head.val + sumList(head.next)
}

File: 03_Linked_List/12_linked_list_sum-list/sum_list.py
class Node:
  def __init__(self, val):
    self.val = val
    self.next = None


# My Iterative Solution

def sum_list(head):
  current_node = head
  sum = 0
  while current_node:
    sum += current_node.val
    current_node = current_node.next
  return sum

# My Recursive Solution

def sum_list(head):
  if head is None:
    return 0
  return head.val + sum_list(head.next)